pinecone/control/pinecone.py

```diff
@@ -222,6 +224,7 @@ def _get_status(self, name: str):
         response = api_instance.describe_index(name)
         return response["status"]
 
-    def Index(self, name: str):
-        index_host = self.index_host_store.get_host(self.index_api, self.config, name)
-        return Index(api_key=self.config.api_key, host=index_host)
+    def Index(self, name: str, host: Optional[str] = None):
```

Comment:
Tiny tweak for supporting custom `host` in grpc, right?

Reply:
Yes, this change is unrelated to `pool_threads` since that only applies to async REST calls, but I wanted to keep these method signatures aligned across both GRPC and non-GRPC usage. I made this change for the REST case so I wouldn't have to figure out how to stub out the call to fetch the host in the test that I added, and in general having more ways to avoid making that network call is a good thing.